utils/code_generator/tiramisu_code_generator.h

```diff
@@ -109,39 +111,51 @@ class buffer {
+//automatically generating a computation
 computation *generate_computation(string name, vector<variable*> computation_variables, int computation_type, vector<computation_abstract*> inputs, vector<int> var_num, int offset);
+//automatically generating computation expression in case of a simple assignment
 string assignment_expression_generator();
+//automatically generating computation expression in case of an assignment using other computations
 string assignment_expression_generator_inputs(vector<computation_abstract*> inputs);
+//automatically generating computation expression in case of stencils
 string stencil_expression_generator(string input_name, computation_abstract* in, vector<int> *var_nums, int offset);
+//returns a vector of computation types according to the probability of their occurence (used in multiple computations codes)
 vector <int> computation_types(int nb_comps, double *probs);
+//returns a vector of padding types according to the probability of their occurence (used in convolutions)
 vector <int> generate_padding_types(int nb_layers, double *padding_probs);
 
 
 
 //=====================================================================tiramisu_code_generator==========================================================================================================
+//automatically generating nb_variales ranging from the from value and initilized with inf_values[i] and sup_values[i]
 vector <variable*> generate_variables(int nb_variables, int from, int *inf_values, int *sup_values);
-
+//automatically generating a single computation tiramisu code with the associated wrapper
 void generate_tiramisu_code_single_computation(int code_id, int computation_type, vector<int> *computation_dims,
                                                vector<int> *var_nums, int nb_inputs, string *default_type_tiramisu,
                                                string *default_type_wrapper, int offset);
+//automatically generating a tiramisu code with convolutional layers
 void generate_tiramisu_code_conv(int code_id, int nb_layers, double *padding_probs, string *default_type_tiramisu, string *default_type_wrapper);
-void generate_tiramisu_layer();
+//automatically generating a multiple computations tiramisu code with the associated wrapper
 void generate_tiramisu_code_multiple_computations(int code_id, vector<int> *computation_dims, int nb_stages, double *probs,
                                                   vector<int> *var_nums, int nb_inputs, string *default_type_tiramisu,
                                                   string *default_type_wrapper, int offset);
 
 //=====================================================================tiramisu_code class==========================================================================================================
 class tiramisu_code{
```

Comment:
Can you add a description for this class ? What does it do generally ? Its name is not very informative of what it does.